### DISCUSSION TITLE
Version 2 Multiplayer and Filestorage

### DISCUSSION THREAD

**Discussion by githubkoma** [tldraw/tldraw-v1#549] [General]
Dear steveruizok,

i just watched with joy your presentation over at BigBlueButtons Youtube Channel https://www.youtube.com/watch?v=4cUaVGIdX1o , especially in regards to Version 2

1)
At one point  you mentioned that Multiplayer/Collaboration will at a point be shipped with an "own solution", instead of liveblocks. 

Will that Multiplayer/Collaboration be free and self-hosted as well? And/Or what will be the specifics or limitations there?

How about implementing yjs like in https://github.com/nimeshnayaju/yjs-tldraw ?

2) 
And will the FileStorage be more flexible/modular? So that perhaps like a Nextcloud Integration could be achieved?

thanks and keep up the great stuff,
githubkoma

**Comment by DaphneMuller** [comment]
Hi, my name is Daphne and I manage Nextcloud's integrations portfolio. We would totally be up for integrating if TLdraw would allow for a free solution for self-hosting. @steveruizok  I'd absolutely love to be in touch with you if this is on the roadmap.

**Reply by steveruizok** [reply]
Hey, I'd be happy to chat: email me at steve@tldraw.com.
